Module Harmony dependency graph (#81)
tc39/proposal-compartments: GRAPH.md

  reference to an image, a URL, or a file system path to the image.
  Incorporating images and similar resources in the JavaScript module graph
  ensures that a program is portable between development and production
  environments, allowing a JavaScript bundler to observe the dependency on an
  image asset.

- JSON modules test the boundary between a language and an asset or resource.
  They do not have dependencies or side-effects.
  Their behavior is much more like decoding than evaluation.

- One valid interpretation of a CSS module would be to incorporate a
  style sheet in the surrounding web page as a side effect of evaluation.
  Such a module would export nothing but allow a bundler to observe the
  dependency and ensure a topological order to the included sheets.
  The module might also export a representation of the generated sheet as a
  mutable resource.

  Style sheets have their own notion of dependencies, and a valid
  interpretation of a CSS module might subvert CSS imports and allow
  CSS modules to benefit from JavaScript module resolution,
  such that a bundler can relocate resources between development and
  production.

> ## Web Assembly not-host-defined modules

Reviewer: ```suggestion
## WebAssembly not-host-defined modules
```

> Not all hosts necessarily implement the web's specifications for allowing

Reviewer: ```suggestion
Not all hosts necessarily implement the web's specifications for allowing
WebAssembly to participate in the JavaScript module graph.
As above, ***a minimal virtual module source protocol*** would allow
user code to emulate the standard behavior for WebAssembly.
```

> Web Assembly to participate in the JavaScript module graph.

Author: ```suggestion
WebAssembly to participate in the JavaScript module graph.
```
